src/astUtils/reflection.ts

```diff
@@ -0,0 +1,193 @@
+import { Statement, Body, AssignmentStatement, Block, ExpressionStatement, CommentStatement, ExitForStatement, ExitWhileStatement, FunctionStatement, IfStatement, IncrementStatement, PrintStatement, GotoStatement, LabelStatement, ReturnStatement, EndStatement, StopStatement, ForStatement, ForEachStatement, WhileStatement, DottedSetStatement, IndexedSetStatement, LibraryStatement, NamespaceStatement, ImportStatement } from '../parser/Statement';
+import { Token } from '../lexer/Token';
+import { LiteralExpression, Expression, BinaryExpression, CallExpression, FunctionExpression, NamespacedVariableNameExpression, DottedGetExpression, XmlAttributeGetExpression, IndexedGetExpression, GroupingExpression, EscapedCharCodeLiteral, ArrayLiteralExpression, AALiteralExpression, UnaryExpression, VariableExpression, SourceLiteralExpression, NewExpression, CallfuncExpression, TemplateStringQuasiExpression, TemplateStringExpression, TaggedTemplateStringExpression } from '../parser/Expression';
+import { BrsType, BrsString, ValueKind, BrsInvalid, BrsBoolean } from '../brsTypes';
+import { BrsNumber } from '../brsTypes/BrsNumber';
+import { BrsFile } from '../files/BrsFile';
+import { XmlFile } from '../files/XmlFile';
+
+// File reflection
+
+export function isBrsFile(file: (BrsFile | XmlFile)): file is BrsFile {
+    return file.extension === '.brs' || file.extension === '.bs';
+}
+
+export function isXmlFile(file: (BrsFile | XmlFile)): file is XmlFile {
+    return file.extension === '.xml';
+}
+
+// Statements reflection
+
+export function isBody(statement: Statement): statement is Body {
+    return statement?.constructor?.name === 'Body';
```

Comment:
I thought we were going to add a `type` property to statements and expressions that contained this info so we didn't have to use `.constructor.name`?

Reply:
I would suggest doing that as part of a separate task - it will be transparent from an API perspective. Do you want me to do it as part of this PR?

Reply:
no, that's a good point. we can do it later.